workshops/build-a-binderhub/workshop-presentations/zero-to-binderhub.md

```diff
@@ -0,0 +1,527 @@
+* `--location` specifies the location of the data centre where your resource will exist.
+  A list of data centre locations can be found [here](https://docs.microsoft.com/en-us/azure/aks/container-service-quotas#region-availability).
+  We have chosen West Europe for resource availability.
+* `--output table` specifies the output should be in human-readable format as opposed to JSON, which is the default output.
+
+### 4. Choose a Cluster Name <a name="aks-step4"></a>
+
+Somewhere on your machine (e.g. `~/Desktop`), create a folder in which to store files relating to the compute cluster we are about to build.
+This folder should have the same name as the cluster and should be descriptive and short.
+
+```bash
+mkdir shfhubcluster
+cd shfhubcluster
+```
+
+> **Discussion topic:** As a team of RSEs managing a BinderHub, where would be the best place for folders such as this to live?
+
+### 5. Create an SSH key <a name="aks-step5"></a>
+
+Create an SSH key to secure your cluster (further details in [this blog post](https://jumpcloud.com/blog/what-are-ssh-keys-b/)). **Keep these files safe.**
+
+```bash
+ssh-keygen -f ssh-key-shfhubcluster
+```
+When prompted for a password, you can choose to leave this blank.
+Some text will be printed to the terminal which you don't need to do anything with.
+
+### 6. Create an Azure Container Service (AKS) Cluster <a name="aks-step6"></a>
+
+This command will request a Kubernetes cluster within the resource group we created.
+It will request one `Standard_D2s_v3` virtual machine which a Kubernetes cluster installed.
+For information on other types of virtual machines available, [see here](https://azure.microsoft.com/en-gb/pricing/details/virtual-machines/series/).
+
+**N.B.:** If you are _not_ using a Free Trial subscription, try setting `--node-count` to **3** instead.
+
+```bash
+az aks create --name shfhubcluster \
+    --resource-group shf_test_hub \
+    --ssh-key-value ssh-key-shfhubcluster.pub \
+    --node-count 1 \
+    --node-vm-size Standard_D2s_v3 \
+    --output table
+```
+* `--name` is the cluster name we defined in [Step 4: Choose a Cluster Name](#aks-step4).
+* `--resource-group` is the resource group we created in [Step 3: Create a Resource Group](#aks-step3).
+* `--ssh-key-value` is the ssh public key we created in [Step 5: Create an SSH key](#aks-step5).
+* `--node-count` is the number of desired nodes in the Kubernetes cluster.
+* `--node-vm-size` is the size of the nodes you wish to use, which varies based on the use-case of the cluster and how much RAM/CPU each user will need.
+
+**N.B.:** The default version of Kubernetes will be installed, you can use the `--kubernetes-version` flag to install a different version.
+
+**This step may take a few minutes to execute.**
+
+### 7. Get credentials from Azure for `kubectl` <a name="aks-step7"></a>
+
+This step automatically updates your Kubernetes client configuration file to be configured with the cluster we've just deployed.
```

Comment:
Maybe elaborate a litte e.g. "this allows kubectl to be 'logged-in' to the kubernetes cluster you just started on azure? 

Might be OK as it is though. It's just that I skipped over the title when I was reading (I don't know how common that is).

Reply:
I'll dig into this a bit more, it's more or less straight from zero-to-jupyterhub.